File: 03_Wed/Cardio.java
import java.util.Scanner;

public class Cardio {
    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);

        System.out.print("String 1: ");
        String input1 = stdin.nextLine();

        System.out.print("String 2: ");
        String input2 = stdin.nextLine();

        // solution1(input1, input2);
        solution2(input1, input2);
        stdin.close();

    }

    public static void solution1(String input1, String input2) {
        String returnedString = "";
        int minLength = returnMinLength(input1, input2);
        for (var i = 0; i < returnConditionLength(input1, input2); i++) {
            if (i < minLength) {
                if (input1.charAt(i) == input2.charAt(i)) {
                    returnedString += input1.charAt(i);
                } else {
                    returnedString += '.';
                }
            } else {
                returnedString += ".";
            }
        }

        System.out.println(returnedString);

    }

    public static void solution2(String input1, String input2) {
        String returnedString = "";
        char[] input1Array = input1.toCharArray();
        char[] input2Array = input2.toCharArray();
        int minLength = returnMinLength(input1, input2);

        int i = 0;
        while (i < returnConditionLength(input1, input2)) {
            if (i < minLength) {
                if (input1Array[i] == input2Array[i]) {
                    returnedString += input1Array[i];
                } else {
                    returnedString += ".";
                }
            }
            else{
                returnedString += ".";
            }


            i++;
        }
        System.out.println(returnedString);

    }

    public static int returnConditionLength(String input, String input2) {
        if (input.length() > input2.length()) {
            return input.length();
        }
        return input2.length();
    }

    public static int returnMinLength(String input, String input2) {
        if (input.length() < input2.length()) {
            return input.length();
        }
        return input2.length();
    }
}

File: 03_Wed/cardio.py
def main():
    input1 = input("String 1: ");
    input2 = input("String 2: ");
    # solution1(input1, input2)
    solution2(input1, input2)

def solution1(input1, input2):
    new_string = ""
    maxLength = returnGreaterLength(input1, input2);
    minLength = returnLesserLength(input1, input2);
    for i in range(0, maxLength):
        if i < minLength:
            if input1[i] == input2[i]:
                new_string += input1[i]
            else:
                new_string += "."
        else:
            new_string += "."

    print(new_string)

def solution2(input1, input2):
    new_string = ""
    maxLength = returnGreaterLength(input1, input2);
    minLength = returnLesserLength(input1, input2);
    i = 0
    while i < maxLength:
        if i < minLength:
            if input1[i] == input2[i]:
                new_string += input1[i]
            else:
                new_string += "."
        else:
            new_string += "."
        i +=1
    print(new_string)


def returnGreaterLength(input1, input2):
    if len(input1) > len(input2):
        return len(input1)
    else:
        return len(input2)

def returnLesserLength(input1, input2):
    if len(input1) < len(input2):
        return len(input1)
    else:
        return len(input2)

    
def mapHandler(new_string,x):
    new_string = "";
    new_string += x
    return new_string


if __name__ == "__main__":
    main()

